hadoop-tools/hadoop-azure/src/test/java/org/apache/hadoop/fs/azurebfs/ITestAzureBlobFileSystemDelete.java

```diff
@@ -130,4 +148,50 @@ public Void call() throws Exception {
     assertPathDoesNotExist(fs, "deleted", dir);
 
   }
+
+  @Test
+  public void testDeleteIdempotency() throws Exception {
+    org.junit.Assume.assumeTrue(DEFAULT_DELETE_CONSIDERED_IDEMPOTENT);
+    // Config to reduce the retry and maxBackoff time for test run
+    AbfsConfiguration abfsConfig = getConfiguration();
+    abfsConfig.setMaxIoRetries(REDUCED_RETRY_COUNT);
+    abfsConfig.setMaxBackoffIntervalMilliseconds(REDUCED_MAX_BACKOFF_INTERVALS_MS);
+
+    final AzureBlobFileSystem fs = getFileSystem();
+    AbfsClient abfsClient = fs.getAbfsStore().getClient();
+    AbfsPerfTracker tracker = new AbfsPerfTracker("test",
+        this.getAccountName(),
+        abfsConfig);
+
+    // Create test AbfsClient
+    AbfsClient testClient = new AbfsClient(
+        abfsClient.getBaseUrl(),
+        new SharedKeyCredentials(abfsConfig.getAccountName().substring(0,
+            abfsConfig.getAccountName().indexOf(DOT)),
+            abfsConfig.getStorageAccountKey()),
+        abfsConfig,
+        new ExponentialRetryPolicy(REDUCED_RETRY_COUNT),
+        abfsConfig.getTokenProvider(),
+        tracker);
+
+    // Mock instance of AbfsRestOperation
+    AbfsRestOperation op = mock(AbfsRestOperation.class);
+    // Set retryCount to non-zero
+    when(op.isARetriedRequest()).thenReturn(true);
+
+    // Mock instance of Http Operation response. This will return HTTP:Not Found
+    AbfsHttpOperation http404Op = mock(AbfsHttpOperation.class);
+    when(http404Op.getStatusCode()).thenReturn(HTTP_NOT_FOUND);
+
+    // Mock delete response to 404
+    when(op.getResult()).thenReturn(http404Op);
+
+    assertEquals(
```

Comment:
These are not asserJ assertions.
That is better alternative, defenitly improved readability. Steve also reccomands the same.
the format is assertThat(<instance on which we are going to make assertion>)
.describedAs("describe what assertion is going tobe made and the same is going to be printed in logs").<assertion method ex: isEqualTo()>

Reply:
Fixed.